src/java/org/apache/velocity/runtime/defaults/velocity.properties

```diff
@@ -262,8 +262,21 @@ introspector.restrict.classes = java.lang.Runtime
 introspector.restrict.classes = java.lang.RuntimePermission
 introspector.restrict.classes = java.lang.SecurityManager
 introspector.restrict.classes = java.lang.System
-introspector.restrict.classes = java.lang.Thread
 introspector.restrict.classes = java.lang.ThreadGroup
 introspector.restrict.classes = java.lang.ThreadLocal
 
+# Restrict instance managers for common servlet containers  (Tomcat, JBoss, Jetty)
+
+introspector.restrict.classes = org.apache.catalina.core.DefaultInstanceManager
+introspector.restrict.classes = org.apache.tomcat.SimpleInstanceManager
+introspector.restrict.classes = org.wildfly.extension.undertow.deployment.UndertowJSPInstanceManager
+introspector.restrict.classes = org.eclipse.jetty.util.DecoratedObjectFactory
+
+# ----------------------------------------------------------------------------
+# SPACE GOBBLING
+# ----------------------------------------------------------------------------
+# Possible values: none, bc (aka Backward Compatible), lines, structured
+# ----------------------------------------------------------------------------
+
+space.gobbling = lines
```

Comment:
This space gobbling block seeems wrong here?!

Reply:
I agree. That's a 2.X feature.